content/intro-to-storybook/react/en/screen.md

```diff
@@ -288,67 +357,89 @@ Storybook's [`play`](https://storybook.js.org/docs/react/writing-stories/play-fu
-+ import { fireEvent, within } from '@storybook/testing-library';
 
-import { PureInboxScreen } from './InboxScreen';
+import InboxScreen from './InboxScreen';
 
-import * as TaskListStories from './TaskList.stories';
+import store from '../lib/store';
+import { rest } from 'msw';
+import { Provider } from 'react-redux';
 
- // A super-simple mock of a redux store
-const Mockstore = configureStore({
-  reducer: {
-    tasks: createSlice({
-      name: 'tasks',
-      initialState: TaskListStories.Default.args.tasks,
-      reducers: {
-        updateTaskState: (state, action) => {
-          const { id, newTaskState } = action.payload;
-          const task = state.findIndex((task) => task.id === id);
-          if (task >= 0) {
-            state[task].state = newTaskState;
-          }
-        },
-      },
-    }).reducer,
-  },
-});
++ import {
++  fireEvent,
++  within,
++  waitFor,
++  waitForElementToBeRemoved
++ } from '@storybook/testing-library';
 
 export default {
-  component: PureInboxScreen,
-  decorators: [story => <Provider store={Mockstore}>{story()}</Provider>],
-  title: 'PureInboxScreen',
+  component: InboxScreen,
+  title: 'InboxScreen',
+  decorators: [(story) => <Provider store={store}>{story()}</Provider>],
 };
 
-const Template = args => <PureInboxScreen {...args} />;
+const Template = () => <InboxScreen />;
 
 export const Default = Template.bind({});
-
-export const Error = Template.bind({});
-Error.args = {
-  error: 'Something',
+Default.parameters = {
+  msw: {
+    handlers: [
+      rest.get(
+        'https://jsonplaceholder.typicode.com/todos?userId=1',
+        (req, res, ctx) => {
+          return res(
+            ctx.json([
+              {
+                id: 1,
+                title: 'Task 1',
+                completed: false,
+              },
+              {
+                id: 2,
+                title: 'Task 2',
+                completed: false,
+              },
+              {
+                id: 3,
+                title: 'Task 3',
+                completed: false,
+              },
+              {
+                id: 4,
+                userID: 2,
+                title: 'Task 4',
+                completed: false,
+              },
+            ])
+          );
+        }
+      ),
+    ],
+  },
 };
 
-+ export const WithInteractions = Template.bind({});
-+ WithInteractions.play = async ({ canvasElement }) => {
++ Default.play = async ({ canvasElement }) => {
 +   const canvas = within(canvasElement);
-+   // Simulates pinning the first task
-+   await fireEvent.click(canvas.getByLabelText("pinTask-1"));
-+   // Simulates pinning the third task
-+   await fireEvent.click(canvas.getByLabelText("pinTask-3"));
++   // Waits for the component to transition from the loading state
++   await waitForElementToBeRemoved(await canvas.findByTestId('loading'));
++   // Waits for the component to be updated based on the store
++   await waitFor(async () => {
```

Comment:
Is the `waitFor` necessary because of the `waitForElementToBeRemoved` before it? Asking because I genuinely don't know. 😅

Reply:
Based on various tests and conversations indeed we have to use this approach to prevent a known issue similar to what I've experienced with the flicker and also to avoid breaking the Interactions if you jump from this story to another and come back. I know it reads a bit "clunky" (pardon the bad pun) but it mitigates the issue and manages to get the stories working.